msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,190 @@
+
+To use nested authentication, follow these steps:
+
+1. [Register their single page application](#register-your-single-page-application)
+1. [Add trusted brokers](#add-trusted-brokers)
+1. [Initialize public client application](#initialize-public-client-application)
+1. [Call an API](#call-an-api)
+
+### Register your single-page application
+
+You need to create a Microsoft Azure App registration for your add-in on the Azure portal. The app registration must have a name, supported account type, and SPA redirect. If your add-in requires additional app registration beyond NAA and SSO, see [Register your single-page application.](/entra/identity-platform/scenario-spa-app-registration)
+
+### Add trusted brokers
+
+To enable nested authentication, your application must actively configure one or more specialized redirect URIs to signal to the Microsoft identity platform that it's allowed to be brokered by supported hosts. The redirect URI of the application must be of type **Single Page Application** and conform to the following scheme:
+
+```
+brk-<broker_application_id>://<your_domain>
+```
+
+Where <broker_application_id> is the app ID or alias of the broker or brokers you wish to trust and <your_domain> is the fully qualified domain name where your application is hosted. For example, **brk-multihub://contoso.com**.
+
+If your app has been upgraded to also run in Outlook and Microsoft365.com (in addition to Teams) then you just need to add one Redirect URI:
+
+```http
+brk-multihub://<your_domain>
+```
+
+Your domain should only include the origin and not sub-paths. For example:
+
+* Correct: brk-multihub://myapp.teams.microsoft.com
+* Incorrect: brk-multihub://myapp.teams.microsoft.com/go
+
+### Initialize public client application
+
+> [!NOTE]
+> Ensure that you initialize TeamsJS before you initialize MSAL (Microsoft Authentication Library), else Nested app authentication might fail.
+
+You need to initialize MSAL and get an instance of the public client application. This is used to get access tokens when needed. We recommended creating the public client application in the `Office.onReady` method.
+
+```javascript
+let pca = undefined;
+
+// Initialize the public client application
+Office.onReady(async (info) => {
+    pca = await msalBrowser.nestablePublicClient.createPublicClientApplication(msalConfig);
+  });
+```
+
+### Acquire your first token
+
+The tokens acquired by MSAL.js through nested app authentication are issued for your Azure app registration ID. The MSAL.js handles token acquisition for user authentication. It tries to get an access token silently, and if that fails, it prompts the user interactively.  The token is then used to call the Microsoft Graph API.
+
+The following steps show the pattern to use for acquiring a token:
+
+1. Call `publicClientApplication.getActiveAccount();` to fetch the active user account from MSAL’s cache.
```

Comment:
here is the updated guidance we are giving 1P developers for acquiring the access token (including getting the account):

The tokens acquired by MSAL.js via NAA will be issued for your app ID. Unlike OBO, you don't need to [preauthorize your hosts](https://review.learn.microsoft.com/en-us/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab?tabs=manifest-teams-toolkit#update-azure-ad-app-registration-for-sso) to call your endpoints. For detailed information on acquiring and using access tokens please see MSAL.js documentation [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/acquire-token.md).  
We recommend first trying the acquireTokenSilent API to try to login the user without any UX interruption. This method requires an active account, so first check for one using getActiveAccount – if there is an active account, use it in your other calls to pca.
If no active account is available, try to retrieve one from cache by calling getAccount with additional filter parameters for tenantID, homeAccountId, and loginHint. These parameters can be gotten from the Teams JS Context interface: [Teams JavaScript client library SDK - Teams | Microsoft Learn](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-library?tabs=javascript%2Cmanifest-teams-toolkit#updates-to-the-context-interface) (note that homeAccountId = userObjectId in Teams JS).  

If no account is available, MSAL will pass an error as InteractionRequiredAuthError and you'll need to use acquireTokenPopup to prompt the user to enter their credentials and complete their sing-in.